pkg/envoy/ads/response_benchmark_test.go

```diff
@@ -0,0 +1,117 @@
+	"github.com/openservicemesh/osm/pkg/configurator"
+	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/envoy"
+	"github.com/openservicemesh/osm/pkg/envoy/registry"
+	configFake "github.com/openservicemesh/osm/pkg/gen/client/config/clientset/versioned/fake"
+	"github.com/openservicemesh/osm/pkg/k8s"
+	"github.com/openservicemesh/osm/pkg/logger"
+	"github.com/openservicemesh/osm/pkg/service"
+	"github.com/openservicemesh/osm/pkg/tests"
+)
+
+func BenchmarkSendXDSResponse(b *testing.B) {
+	b.StopTimer()
+	if err := logger.SetLogLevel("error"); err != nil {
+		b.Logf("Failed to set log level to error: %s", err)
+	}
+
+	var (
+		mockCtrl         *gomock.Controller
+		mockConfigurator *configurator.MockConfigurator
+	)
+
+	mockCtrl = gomock.NewController(b)
+	mockConfigurator = configurator.NewMockConfigurator(mockCtrl)
+	fakeCertManager, err := certificate.FakeCertManager()
+	if err != nil {
+		b.Fatalf("Failed to create fake cert manager: %s", err)
+	}
+
+	// --- setup
+	namespace := tests.Namespace
+
+	kubeClient := testclient.NewSimpleClientset()
+	configClient := configFake.NewSimpleClientset()
+	proxyService := service.MeshService{
+		Name:      tests.BookstoreV1ServiceName,
+		Namespace: namespace,
+	}
+	proxySvcAccount := tests.BookstoreServiceAccount
+
+	mockConfigurator.EXPECT().GetCertKeyBitSize().Return(2048).AnyTimes()
+
+	proxyUUID := uuid.New()
+	labels := map[string]string{constants.EnvoyUniqueIDLabelName: proxyUUID.String()}
+	mc := fakeCatalog.NewFakeMeshCatalog(kubeClient, configClient)
+	proxyRegistry := registry.NewProxyRegistry(registry.ExplicitProxyServiceMapper(func(*envoy.Proxy) ([]service.MeshService, error) {
+		return nil, nil
+	}), nil)
+
+	pod := tests.NewPodFixture(namespace, fmt.Sprintf("pod-0-%s", proxyUUID), tests.BookstoreServiceAccountName, tests.PodLabels)
+	pod.Labels[constants.EnvoyUniqueIDLabelName] = proxyUUID.String()
+	_, err = kubeClient.CoreV1().Pods(namespace).Create(context.TODO(), &pod, metav1.CreateOptions{})
+	if err != nil {
+		b.Fatalf("Failed to create pod: %v", err)
+	}
+
+	svc := tests.NewServiceFixture(proxyService.Name, namespace, labels)
+	_, err = kubeClient.CoreV1().Services(namespace).Create(context.TODO(), svc, metav1.CreateOptions{})
+	if err != nil {
+		b.Fatalf("Failed to create service: %v", err)
+	}
+
+	certManager := fakeTresor.NewFake(nil)
+	certCommonName := envoy.NewXDSCertCommonName(proxyUUID, envoy.KindSidecar, proxySvcAccount.Name, proxySvcAccount.Namespace)
+	certDuration := 1 * time.Hour
+	certPEM, _ := certManager.IssueCertificate(certCommonName, certDuration)
+	cert, _ := certificate.DecodePEMCertificate(certPEM.GetCertificateChain())
+	server, _ := tests.NewFakeXDSServer(cert, nil, nil)
+	kubectrlMock := k8s.NewMockController(mockCtrl)
+	certSerialNumber := certificate.SerialNumber("123456")
+	proxy, _ := envoy.NewProxy(certCommonName, certSerialNumber, nil)
+
```

Comment:
I'm wary about mocking function calls in benchmarks. Could we just use a fake clientset with real clients?

Reply:
This won't be an issue here. On line 109, the benchmark timer is reset, thus the setup time is not included.

Reply:
I think keith’s concern is more that when the timer starts the mocks and fakes use a lot of bogus code paths. 

We should be able to benchmark everything , only use fake client sets to avoid network calls

Reply:
Replacing mocks with actual components. Please check again